backend/api/views.py

```diff
@@ -103,4 +106,23 @@ def uptake(request):
     """
     Take a JSON payload from telemetry streaming to record experiment uptake.
     """
+    payload = request.data.get('uptake')
+    if not payload or not isinstance(payload, list):
+        raise ValidationError
+
+    payload = payload[0]
```

Comment:
Won't they post an array of objects describing each experiment/branch pair?  Should we loop here?

Reply:
Yes but for now each request will only have 1 object in the array. But you're right, even so, a loop would be better. Thanks!